### ISSUE TITLE
I cannot find any way to abort running kernel!

### ISSUE THREAD

**Issue by AbedKarmi** [m4rs-mt/ILGPU#792]
I am running a kernel with a very time consuming process, I could not find a way to interact with it, or to abort it!. Is there a way to abort/kill the running kernel? or a way to interact with running process, so we can let it finish the loop by changing a status of a varialbe ?

**Comment by MoFtZ** [comment]
hi @AbedKarmi. I have never tried to abort a kernel, but you might be able to adapt the MonitorProgress sample code to achieve your desired behaviour.

In the sample, the kernel updates a variable that can be read by the CPU. I'm assuming that it would be possible to do the reverse - update a variable from the CPU and read it from the kernel.

https://github.com/m4rs-mt/ILGPU/blob/master/Samples/MonitorProgress/Program.cs

Let us know how you go.

**Comment by AbedKarmi** [comment]
Great Thanks, it just worked :)

**Comment by AbedKarmi** [comment]
That worked for Cuda device (Nvidia Quadro RTX 4000). It cannot be used for an Intel card!. Still need a way to abort the process or kill the kernel.

**Comment by MoFtZ** [comment]
What do you mean by an Intel card? An iGPU running OpenCL?

**Comment by AbedKarmi** [comment]
Yes, that is what I mean. The method  worked for Quadro RTX4000 which is Cuda type. Look what it does :

searches in :  context.GetCudaDevices()
Creates Cudat Accelerator : device.CreateCudaAccelerator(context)
Creates a variable :  using var progress = new CudaProgress<ulong>(accelerator);

All that is not applicable for an OponCL !. Intel graphics card type is OpenCL.

**Comment by AbedKarmi** [comment]
I tried to modify the code that allocates buffer to this  :

  CLException.ThrowIfFailed(CLAPI.CurrentAPI.CreateBuffer(accelerator.NativePtr,CLBufferFlags.CL_MEM_READ_WRITE, new IntPtr(LengthInBytes),IntPtr.Zero, out IntPtr resultPtr));

but the kernel throws an exception when it is called!

**Comment by m4rs-mt** [comment]
@AbedKarmi Personally, I would *not* recommend trying to abort running kernels, as they are meant to work asynchronously on another device independently. I would recommend splitting the workload into multiple smaller kernels or setting up different grid configurations to process different chunks separately, as this is generally much more reliable.

**Comment by MoFtZ** [comment]
@AbedKarmi
As far as I can tell, there is no way to achieve your desired behavior using OpenCL.

I would suggest you redesign your kernel to allow it to continue from a previous iteration. For example, instead of running indefinitely, run for 1,000 iterations. When the kernel completes, your program checks if it should launch the kernel for another 1,000 iterations.

**Comment by m4rs-mt** [comment]
We are closing this issue for now. If the general problem persists, we can safely reopen this issue.
